compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -378,6 +378,13 @@ private void visitFunction(BLangFunction funcNode) {
         if (isPublicInvokableNode(funcNode)) {
             analyzeNode(funcNode.returnTypeNode, invokableEnv);
         }
+        for (BLangSimpleVariable requiredParam : funcNode.requiredParams) {
+            analyzeNode(requiredParam, env);
```

Comment:
Since the env of the params of a function must be the env inside the function, the second param should be `invokableEnv`